blog_post_collaboration/Integrating_Kedro_with_Databricks.md

```diff
@@ -0,0 +1,136 @@
+        with open(Path.home() / ".databrickscfg") as f:
+            lines = f.readlines()
+
+        idx = 0
+        default_idx = 0
+        default_found = False
+        for line in lines:
+            idx = idx + 1
+            if f"[{DEFAULT}]" in line:
+                default_idx = idx
+                default_found = True
+            elif "[" in line:
+                if default_found == False:
+                    continue
+                break
+
+
+        out = "".join(lines[default_idx:idx]).split("\n")
+        for line in out:
+            if "host" in line:
+                host = line.split("=", 1)[1].split("//", 1)[1].strip()[:-1]
+            if "cluster_id" in line:
+                cluster_id = line.split("=", 1)[1].strip()
+            if "token" in line:
+                token = line.split("=", 1)[1].strip()
+
+        os.environ[
+            "SPARK_REMOTE"
+        ] = f"sc://{host}:443/;token={token};x-databricks-cluster-id={cluster_id}"
+ 
+``` 
+
+This example will populate SPARK_REMOTE with your local databrickscfg file. We do't setup the remote connection if the project is being run from inside Databricks (if SPARK_HOME points to Databricks), so you're still able to run it in the usual [hybrid development flow](https://docs.kedro.org/en/stable/deployment/databricks/databricks_ide_development_workflow.html).
+Notice that we don’t need to setup a `spark.yml` file as is common in other PySpark templates; we’re not passing any configuration, just using the cluster that is in Databricks. We also don’t need to load any extra Spark files (e.g. JARs), as we are using a thin Spark Connect client.
+ 
+Now all your Spark calls in your pipelines will automatically use the remote cluster. There's no need to change anything in your code. However, notebooks might be part of the project. To use your remote cluster without needing to use environment variables, you can use the DatabricksSession:
+```
+from databricks.connect import DatabricksSession
+ 
+spark = DatabricksSession.builder.getOrCreate()
+```
+This will read your configuration directly from the `databrickscfg` file, bypassing the setup needed for pure Spark Connect.
+
+When using the remote cluster, it's preferred to avoid data transfers between the environments, with all catalog entries referencing remote locations. Using ` kedro_datasets.databricks.ManagedTableDataSet` as your dataset type in the catalog also allows you use Delta table features.
+
+
+ 
+## Enabling MLFlow on Databricks
+Using [MLflow](https://mlflow.org/) to save all your artifacts directly to Databricks leads to a powerful workflow. For this we can use [`kedro-mlflow`](https://github.com/Galileo-Galilei/kedro-mlflow). Note that `kedro-mlflow` is built on top of the mlflow library and although the databricks config cannot be found in its documentation, you can read more about it in the [documentation from mlflow directly](https://mlflow.org/docs/latest/index.html).

```

Comment:
Is there a reason not using `kedro mlflow init` but creating config manually?

Reply:
You can use `kedro mlflow init` to generate the mlflow.yml file, the documentation of `kedro-mlflow` just doesn't mention you can configure your server to point to `databricks`. This info is in the `mlflow` docs